extensions/amp-bind/0.1/test/test-bind-integration-ptag.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createFixtureIframe} from '../../../../testing/iframe';
+import {bindForDoc} from '../../../../src/bind';
+import {ampdocServiceFor} from '../../../../src/ampdoc';
+
+describe.configure().retryOnSaucelabs().run('amp-bind', function() {
+  const fixtureLocation = 'test/fixtures/amp-bind-integrations/p-tag.html';
+
+  let fixture;
+  let ampdoc;
+
+  this.timeout(5000);
+
+  beforeEach(() => {
+    return createFixtureIframe(fixtureLocation).then(f => {
+      fixture = f;
+      return waitForEvent('amp:bind:initialize');
+    }).then(() => {
+      const ampdocService = ampdocServiceFor(fixture.win);
+      ampdoc = ampdocService.getAmpDoc(fixture.doc);
+    });
+  });
+
+ /**
+  * @param {string} name
+  * @return {!Promise}
+  */
+  function waitForEvent(name) {
+    return new Promise(resolve => {
+      function callback() {
+        resolve();
+        fixture.win.removeEventListener(name, callback);
+      };
+      fixture.win.addEventListener(name, callback);
+    });
+  }
+
+ /** @return {!Promise} */
+  function waitForBindApplication() {
+   // Bind should be available, but need to wait for actions to resolve
+   // service promise for bind and call setState.
+    return bindForDoc(ampdoc).then(unusedBind =>
+       waitForEvent('amp:bind:setState'));
+  }
+
+  describe('text integration', () => {
```

Comment:
Let's remove this describe block and put the tests directly under the top-level describe.